src/users/user.rs

```diff
@@ -0,0 +1,148 @@
+use std::collections::HashMap;
+use snafu::ensure;
+use pwhash::bcrypt;
+use uuid::Uuid;
+use serde::{Serialize, Deserialize};
+
+use crate::error::Result;
+use crate::error;
+use core::fmt;
+use std::str::FromStr;
+
+#[derive(Debug, PartialEq, Eq, Serialize, Deserialize, Clone, Hash)]
+pub struct UserRegistration {
+    pub email: String,
+    pub password: String,
+    pub real_name :String,
+}
+
+#[derive(Debug, PartialEq, Eq, Serialize, Deserialize, Clone, Hash)]
+pub struct UserCredentials {
+    pub email: String,
+    pub password: String,
+}
+
+#[derive(Debug, PartialEq, Eq, Serialize, Deserialize, Clone, Hash)]
+pub struct UserIdentification {
+    id: Uuid,
+}
+
+impl UserIdentification {
+    fn new() -> Self {
+        Self { id: Uuid::new_v4() }
```

Comment:
will this be parametrized in the future? Maybe you can implement Default

Reply:
imho new is semantically better here. default  implies to me that it always returns the same value